File: davinci-frontend/src/lib/types/job.ts
export interface IngestionJob {
  id: string;
  project: string;
  job_type: 'pubmed_search' | 'pubmed_fetch' | 'geo_search' | 'sra_search' | 'variant_annotation' | 'gene_ner';
  status: 'pending' | 'running' | 'completed' | 'failed' | 'cancelled';
  parameters: Record<string, unknown>;
  records_processed: number;
  records_inserted: number;
  records_updated: number;
  error_message: string | null;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
}

File: apps/core/serializers/job.py
from rest_framework import serializers
from apps.core.models import IngestionJob


class IngestionJobSerializer(serializers.ModelSerializer):
    class Meta:
        model = IngestionJob
        fields = [
            'id', 'job_type', 'status', 'parameters',
            'records_processed', 'records_inserted', 'records_updated',
            'error_message', 'created_at', 'started_at', 'completed_at',
        ]
        read_only_fields = fields


class JobCancelErrorSerializer(serializers.Serializer):
    """Resposta de erro quando o job já está em estado terminal."""
    detail = serializers.CharField()
